js_modules/dagster-ui/packages/ui-core/src/pipelines/PipelineExplorerRoot.tsx

```diff
@@ -88,6 +89,7 @@ export const PipelineExplorerContainer = ({
       : explorerPath.snapshotId
       ? explorerPath.snapshotId
       : undefined;
+
     // only add snapshot id from workspace if its loaded when we first render
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [explorerPath]);
```

Comment:
Another way to fix this would have been to add `explorerPath.snapshotId` as the specific dep  (in case  other fields of the object change with the snapshotId staying the same) but I see no reason not to also memoize explorerPath

Reply:
ill do both and we can delete it all when react compiler saves us from ourselves

Reply:
@alangenfeld I don't think React compiler can really detect this case. It would by default memoize by everything, in this case we want everything except the snapshotIdFromContext